app/workers/internal_worker.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class InternalWorker
+  include Sidekiq::Worker
+  attr_reader :receiver
+
+  def perform(id)
+    @receiver = Receiver.find_by(id)
+    return unless receiver
+    dispatch_events
+    InternalWorker.perform_at(receiver.interval.from_now, id)
```

Comment:
Same thing here about scheduling it based on the start time, the interval, and the smallest `n` that puts it in the future